cpp/src/arrow/array/builder_nested.h

```diff
@@ -202,6 +201,11 @@ class BaseListBuilder : public ArrayBuilder {
     const int64_t num_values = value_builder_->length();
     return offsets_builder_.Append(static_cast<offset_type>(num_values));
   }
+
+  void UnsafeAppendNextOffset() {
```

Comment:
Minor nit, but perhaps we could use this in `AppendNulls` and `AppendEmptyValues`.

Reply:
I'm not sure. In `AppendNulls` and `AppendEmptyValues` the same offset is appended multiple times. Currently this number is only computed once. If we use `UnsafeAppendNextOffset` there, the line `const int64_t num_values = value_builder_->length();` will be computed multiple times.